src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CppCodegenNodeFactory.cs

```diff
@@ -38,5 +39,26 @@ protected override ISymbolNode CreateReadyToRunHelperNode(Tuple<ReadyToRunHelper
             // TODO: this is wrong: this returns an assembly stub node
             return new ReadyToRunHelperNode(helperCall.Item1, helperCall.Item2);
         }
+
+        public override void AttachToDependencyGraph(DependencyAnalyzerBase<NodeFactory> graph)
+        {
+            base.AttachToDependencyGraph(graph);
+
+            var rooting = new RootingServiceProvider(graph, this);
+
+            foreach (var wkt in Enum.GetValues(typeof(WellKnownType)))
```

Comment:
Why do we need to root these from here? Can we make this work implicitly? (I.e. only when the type is actually used we will generate the EEType for it.)